src/components/Game/Game.jsx

```diff
@@ -0,0 +1,162 @@
+        this.enableBotModeHandler = this.enableBotMode.bind(this);
+        this.disableBotModeHandler = this.disableBotMode.bind(this)
+        this.resetScoreHandler = this.resetScore.bind(this);
+    }
+
+    componentDidMount() {
+        const data = getLocalStorageData(LocalStorageKeys.game);
+        if (data) {
+            this.setState(data);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (!this.state.isGameOver) {
+            const lastSquares = this.state.history[this.state.history.length - 1].squares;
+            const winner = calculateWinner(lastSquares);
+            if (winner) {
+                this.setState({
+                    isGameOver: true,
+                    score: {
+                        winsX: winner === GAME_SYMBOLS.X ? this.state.score.winsX + 1 : this.state.score.winsX,
+                        winsO: winner === GAME_SYMBOLS.O ? this.state.score.winsO + 1 : this.state.score.winsO,
+                    }
+                })
+            } else if (lastSquares.indexOf(null) < 0) {
+                this.setState({
+                    isGameOver: true,
+                })
+            }
+        }
+        if (prevState !== this.state) {
+            updateLocalStorageData(LocalStorageKeys.game, this.state);
+        }
+    }
+
+    initStartState() {
+        return {
+            history: [{
+                squares: Array(9).fill(null),
+            }],
+            xIsNext: true,
+            stepNumber: 0,
+            botMode: false,
+            isGameOver: false,
+            isMoveBlocked: false,
+            score: {
+                winsX: 0,
+                winsO: 0,
+            }
+        }
+    }
+
+    fillSquare(i) {
+        const history = this.state.history.slice(0, this.state.stepNumber + 1);
+        const currentSquares = history[history.length - 1].squares;
+        if (!(calculateWinner(currentSquares) || currentSquares[i])) {
+            const squares = [...currentSquares];
+            squares[i] = this.state.xIsNext ? GAME_SYMBOLS.X : GAME_SYMBOLS.O;
+            this.setState(() => {
+                return {
+                    history: history.concat([{squares: squares}]),
+                    xIsNext: !this.state.xIsNext,
+                    stepNumber: history.length,
+                }
+            });
+        }
+    }
+
+
+    handleClick(i) {
+        if (!(this.state.isMoveBlocked || this.state.isGameOver)) {
+            this.fillSquare(i);
+            const squares = this.state.history[this.state.stepNumber].squares;
+            const isMoveLast = squares.filter(el => el === null).length === 1;
+            if (this.state.botMode && !(isMoveLast || squares[i])) {
+                this.setState({isMoveBlocked: true});
+                setTimeout(() => {
+                    this.setState({isMoveBlocked: false});
+                    this.fillSquare(bestBotMove(this.state.history[this.state.history.length - 1].squares));
```

Comment:
create a variable for `bestBotMove` result

Reply:
```suggestion
                 setTimeout(() => {
                    const botMoveIndex = calculateBotMove(this.state.history[this.state.history.length - 1].squares);
                    this.setState({isMoveBlocked: false});
                    this.fillSquare(botMoveIndex);
                }, 500)
```
Fixed